docs/user/implement_algo.md

```diff
@@ -0,0 +1,635 @@
+# Implement a New Algorithm
+
+In this section, we will describe how to implement an RL algorithm using garage.
+Note that this section assumes some level of familiarity with reinforcement
+learning. For a more gentle introduction to the field of reinforcement learning
+as a whole, we recommend consulting [OpenAI's Spinning Up](https://spinningup.openai.com/en/latest/user/introduction.html).
+
+```eval_rst
+We will start by introducing the core :code:`RLAlgorithm` API used in garage,
+then show how to implement the classical REINFORCE :cite:`williams1992simple`
+algorithm, also known as the "vanilla" policy gradient (VPG).
+```
+
+## Algorithm API
+
+All RL algorithms used with garage implement a small interface that allows
+accessing important services such as snapshotting, "plotting" (visualization of
+the current policy in the environment), and resume.
+
+The interface requires a single method, `train(runner)`, which takes a
+`garage.experiment.LocalRunner`. The interface is defined in
+`garage.np.algos.RLAlgorithm`, but inheriting from this class isn't necessary.
+
+Some additional functionality (such as sampling and plotting) require additional
+fields to exist.
+
+```py
+class RLAlgorithm(abc.ABC):
```

Comment:
can you use a literal include for this? 

see https://github.com/rlworkgroup/garage/pull/1765/files#diff-f6fdd52adff0a248b20cbb76973f87e0R42 for an example